app/javascript/components/rooms/room/join/JoinCard.jsx

```diff
@@ -96,7 +97,12 @@ export default function JoinCard() {
     } else if (currentUser?.name) {
       methods.setValue('name', currentUser.name);
     }
-  }, [joinFormName, currentUser?.name]);
+
+    // Default viewerCode if passed as query params
+    if (viewerCode) {
+      methods.setValue('access_code', viewerCode);
+    }
+  }, [viewerCode, joinFormName, currentUser?.name]);
```

Comment:
I would put this in a separate useEffect just for cleanliness

Reply:
Okay, updated!